californium-core/src/main/java/org/eclipse/californium/core/network/stack/Block2BlockwiseStatus.java

```diff
@@ -0,0 +1,302 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations - initial creation
+ ******************************************************************************/
+package org.eclipse.californium.core.network.stack;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.californium.core.coap.BlockOption;
+import org.eclipse.californium.core.coap.MessageObserverAdapter;
+import org.eclipse.californium.core.coap.OptionSet;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.network.Exchange;
+
+
+/**
+ * A tracker for the blockwise transfer of a response body.
+ *
+ */
+final class Block2BlockwiseStatus extends BlockwiseStatus {
+
+	private static final Logger LOGGER = Logger.getLogger(Block2BlockwiseStatus.class.getName());
+
+	private Response response;
+	private byte[] etag;
+
+	/**
+	 * @param maxSize
+	 * @param contentFormat
+	 */
+	private Block2BlockwiseStatus(final int contentFormat) {
```

Comment:
The currently caller `OutboundResponse` may better call the other constructor to use the allocate of the base class.
And the canditate for this `forRandomAccessRequest` requires the other 2 parameters.